src/NHibernate/Async/AdoNet/MySqlClientBatchingBatcher.cs

```diff
@@ -0,0 +1,80 @@
+﻿//------------------------------------------------------------------------------
+// <auto-generated>
+//     This code was generated by AsyncGenerator.
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+//------------------------------------------------------------------------------
+
+
+using System;
+using System.Data.Common;
+using System.Text;
+using NHibernate.AdoNet.Util;
+using NHibernate.Exceptions;
+
+namespace NHibernate.AdoNet
+{
+	using System.Threading.Tasks;
+	public partial class MySqlClientBatchingBatcher : AbstractBatcher
+	{
+
+		public override async Task AddToBatchAsync(IExpectation expectation)
+		{
+			totalExpectedRowsAffected += expectation.ExpectedRowCount;
+			var batchUpdate = CurrentCommand;
+			await (PrepareAsync(batchUpdate)).ConfigureAwait(false);
+			Driver.AdjustCommand(batchUpdate);
+			string lineWithParameters = null;
+			var sqlStatementLogger = Factory.Settings.SqlStatementLogger;
+			if (sqlStatementLogger.IsDebugEnabled || Log.IsDebugEnabled)
+			{
+				lineWithParameters = sqlStatementLogger.GetCommandLineWithParameters(batchUpdate);
+				var formatStyle = sqlStatementLogger.DetermineActualStyle(FormatStyle.Basic);
+				lineWithParameters = formatStyle.Formatter.Format(lineWithParameters);
+				currentBatchCommandsLog.Append("command ")
+					.Append(currentBatch.CountOfCommands)
+					.Append(":")
+					.AppendLine(lineWithParameters);
+			}
+			if (Log.IsDebugEnabled)
+			{
+				Log.Debug("Adding to batch:" + lineWithParameters);
+			}
+			currentBatch.Append(batchUpdate);
+
+			if (currentBatch.CountOfCommands >= batchSize)
+			{
+				await (DoExecuteBatchAsync(batchUpdate)).ConfigureAwait(false);
+			}
+		}
+
+		protected override async Task DoExecuteBatchAsync(DbCommand ps)
+		{
+			Log.DebugFormat("Executing batch");
+			await (CheckReadersAsync()).ConfigureAwait(false);
+			if (Factory.Settings.SqlStatementLogger.IsDebugEnabled)
+			{
+				Factory.Settings.SqlStatementLogger.LogBatchCommand(currentBatchCommandsLog.ToString());
+				currentBatchCommandsLog = new StringBuilder().AppendLine("Batch commands:");
+			}
+
+			int rowsAffected;
+			try
+			{
+				rowsAffected = currentBatch.ExecuteNonQuery();
```

Comment:
The method `DoExecuteBatch` from `MySqlClientBatchingBatcher` executes `ExecuteNonQuery` method from `MySqlClientSqlCommandSet` which executes a delegate created by `DelegateHelper.BuildFunc<int>(adapterType, "ExecuteBatch")`. 
The generator is not able to transform this.